R/target_market_share.R

```diff
@@ -401,3 +407,49 @@ aggregate_by_loan_id <- function(data) {
     ) %>%
     ungroup()
 }
+
+check_unexpected_columns <- function(data) {
+
+  possible_matched_columns <- c(
+    "id_loan",
+    "id_direct_loantaker",
+    "name_direct_loantaker",
+    "id_intermediate_parent_1",
+    "name_intermediate_parent_1",
+    "id_ultimate_parent",
+    "name_ultimate_parent",
+    "loan_size_outstanding",
+    "loan_size_outstanding_currency",
+    "loan_size_credit_limit",
+    "loan_size_credit_limit_currency",
+    "sector_classification_system",
+    "sector_classification_input_type",
+    "sector_classification_direct_loantaker",
+    "fi_type",
+    "flag_project_finance_loan",
+    "name_project",
+    "lei_direct_loantaker",
+    "isin_direct_loantaker",
+    "id_2dii",
+    "level",
+    "sector",
+    "sector_ald",
+    "name",
+    "name_ald",
+    "score",
+    "source",
+    "borderline"
+    )
+
+unexpected_names <- setdiff(names(data), possible_matched_columns)
```

Comment:
How about `invalid_columns`? The word invalid more strongly conveys the program will fail -- as opposed to "unexpected"  (e.g. a present might be unexpected but welcome).